physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,848 @@
+!***********************************************************************
+!> TODO: replace with appropriate licence for CCPP
+!*   GNU Lesser General Public License
+!*   <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @brief Land IAU (Incremental Analysis Update) module, 
+!> for the NoahMP soil/snow temperature (can be extended to include soil moisture) 
+
+!! \section land_iau_mod
+!> - reads settings from namelist file (which indicates if IAU increments are available or not)
+!> - reads in DA increments from GSI/JEDI DA at the start of (the DA) cycle 
+!> - maps increments to FV3 grid points belonging to mpi process
+!> - interpolates temporally (with filter-weights if required by configuration)
+!> - updates states with the interpolated increments
+
+!> March, 2024: Tseganeh Z. Gichamo, (EMC) based on the FV3 IAU mod 
+!> by Xi.Chen <xi.chen@noaa.gov> and Philip Pegion, PSL <philip.pegion@noaa.gov>
+!-------------------------------------------------------------------------------
+
+module land_iau_mod
+  
+  use machine,                  only: kind_phys, kind_dyn
+  use physcons,                 only: pi => con_pi

```

Comment:
I removed this in https://github.com/tsga/ccpp-physics/pull/1 since `pi` isn't used and this isn't how physical constants are supposed to be passed in CCPP schemes.